compound/src/main/kotlin/io/element/android/compound/theme/MaterialThemeColors.kt

```diff
@@ -16,83 +16,88 @@
-    onPrimaryContainer = LightColorTokens.colorGray1400,
-    inversePrimary = LightColorTokens.colorThemeBg,
-    secondary = LightColorTokens.colorGray900,
-    onSecondary = LightColorTokens.colorThemeBg,
-    secondaryContainer = LightColorTokens.colorGray400,
-    onSecondaryContainer = LightColorTokens.colorGray1400,
-    tertiary = LightColorTokens.colorGray900,
-    onTertiary = LightColorTokens.colorThemeBg,
-    tertiaryContainer = LightColorTokens.colorGray1400,
-    onTertiaryContainer = LightColorTokens.colorThemeBg,
-    background = LightColorTokens.colorThemeBg,
-    onBackground = LightColorTokens.colorGray1400,
-    surface = LightColorTokens.colorThemeBg,
-    onSurface = LightColorTokens.colorGray1400,
-    surfaceVariant = LightColorTokens.colorGray300,
-    onSurfaceVariant = LightColorTokens.colorGray900,
-    surfaceTint = LightColorTokens.colorGray1000,
-    inverseSurface = LightColorTokens.colorGray1300,
-    inverseOnSurface = LightColorTokens.colorThemeBg,
-    error = LightColorTokens.colorRed900,
-    onError = LightColorTokens.colorThemeBg,
-    errorContainer = LightColorTokens.colorRed400,
-    onErrorContainer = LightColorTokens.colorRed900,
-    outline = LightColorTokens.colorGray800,
-    outlineVariant = LightColorTokens.colorAlphaGray400,
-    scrim = LightColorTokens.colorGray1400,
-)
-
-@OptIn(CoreColorToken::class)
-internal val materialColorSchemeDark = darkColorScheme(
-    primary = DarkColorTokens.colorGray1400,
-    onPrimary = DarkColorTokens.colorThemeBg,
-    primaryContainer = DarkColorTokens.colorThemeBg,
-    onPrimaryContainer = DarkColorTokens.colorGray1400,
-    inversePrimary = DarkColorTokens.colorThemeBg,
-    secondary = DarkColorTokens.colorGray900,
-    onSecondary = DarkColorTokens.colorThemeBg,
-    secondaryContainer = DarkColorTokens.colorGray400,
-    onSecondaryContainer = DarkColorTokens.colorGray1400,
-    tertiary = DarkColorTokens.colorGray900,
-    onTertiary = DarkColorTokens.colorThemeBg,
-    tertiaryContainer = DarkColorTokens.colorGray1400,
-    onTertiaryContainer = DarkColorTokens.colorThemeBg,
-    background = DarkColorTokens.colorThemeBg,
-    onBackground = DarkColorTokens.colorGray1400,
-    surface = DarkColorTokens.colorThemeBg,
-    onSurface = DarkColorTokens.colorGray1400,
-    surfaceVariant = DarkColorTokens.colorGray300,
-    onSurfaceVariant = DarkColorTokens.colorGray900,
-    surfaceTint = DarkColorTokens.colorGray1000,
-    inverseSurface = DarkColorTokens.colorGray1300,
-    inverseOnSurface = DarkColorTokens.colorThemeBg,
-    error = DarkColorTokens.colorRed900,
-    onError = DarkColorTokens.colorThemeBg,
-    errorContainer = DarkColorTokens.colorRed400,
-    onErrorContainer = DarkColorTokens.colorRed900,
-    outline = DarkColorTokens.colorGray800,
-    outlineVariant = DarkColorTokens.colorAlphaGray400,
-    scrim = DarkColorTokens.colorGray300,
-)
+fun SemanticColors.toMaterialColorScheme(): ColorScheme {
+    return if (isLight) {
```

Comment:
Thanks for trying to simplify this, but given [the figma file](https://www.figma.com/design/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?node-id=311-14&t=2moYxYSKg0a2Lhnn-1) we use as the source for these I think we should probably try to use the tokens directly.

If you think it's better to use the semantic tokens we should double check all of them (i.e. `primaryContainer` is supposed to be `color.bg.canvas.default`).

Reply:
Yes. The idea of this extension is to be able to have one single source of truth: `SemanticColors`. Else if we want to customize the colors, we will have to provide both SemanticColors and MaterialColor. I am sure we can do a mapping between the two, but I agree that the work is not over, and it's probably because the current values were not correct.